File: Application_part/project_final_version/app/src/main/java/com/example/heartbeat/ViewHolder.java
package com.example.heartbeat;

import android.view.View;
import android.widget.TextView;

import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

public class ViewHolder extends RecyclerView.ViewHolder{
    //LinearLayout linearLayout;
    CardView cardView;
    TextView DeviceTextView;
    TextView DeviceMACTextView;

    ViewHolder(View itemView, BLEDataAdapter.ItemClickListener listener){
        super(itemView);

        DeviceTextView = itemView.findViewById(R.id.device_item_name);
        DeviceMACTextView = itemView.findViewById(R.id.device_item_mac);
        cardView = itemView.findViewById(R.id.card_view);

        cardView.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                int position = getAdapterPosition();

                if(position != RecyclerView.NO_POSITION){
                    listener.onItemClick(v, position);
                }
            }
        });
        /*
        itemView.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                int position = getAdapterPosition();

                if(position != RecyclerView.NO_POSITION){
                    listener.onItemClick(v, position);
                }
            }
        });
*/

    }
}

File: Application_part/project_kotlin_version/Kotlin_ver/app/src/main/java/com/example/kotlin_ver/ViewHolder.kt
package com.example.kotlin_ver

import android.view.View
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView


class ViewHolder(itemView : View, listener : BLEDataAdapter.ItemClickListener) : RecyclerView.ViewHolder(itemView) {
    val deviceTextView : TextView = itemView.findViewById(R.id.device_item_name)
    val deviceMACTextView : TextView = itemView.findViewById(R.id.device_item_mac)
    val cardView : CardView = itemView.findViewById(R.id.card_view)
    val listener = listener

    fun bind() {
        cardView.setOnClickListener(
            object : View.OnClickListener {
                override fun onClick(v: View) {
                    val position: Int = getAdapterPosition()
                    if (position != RecyclerView.NO_POSITION) {
                        listener.onItemClick(v, position)
                    }
                }
            })
    }

}